src/workerd/api/node/zlib-util.h

```diff
@@ -234,11 +240,43 @@ class ZlibUtil final: public jsg::Object {
     }
   };
 
-  uint32_t crc32Sync(kj::Array<kj::byte> data, uint32_t value);
+  struct Options {
+    jsg::Optional<int> flush;
+    jsg::Optional<int> finishFlush;
+    jsg::Optional<int> chunkSize;
+    jsg::Optional<kj::uint> windowBits;
+    jsg::Optional<int> level;
+    jsg::Optional<kj::uint> memLevel;
+    jsg::Optional<kj::uint> strategy;
+    jsg::Optional<kj::Array<kj::byte>> dictionary;
+    // We'll handle info on the JS side for now
+    jsg::Optional<kj::uint> maxOutputLength;
+
+    JSG_STRUCT(flush,
+        finishFlush,
+        chunkSize,
+        windowBits,
+        level,
+        memLevel,
+        strategy,
+        dictionary,
+        maxOutputLength);
+  };
+
+  using InputSource = kj::OneOf<jsg::NonCoercible<kj::String>, kj::Array<kj::byte>>;
+  using CompressCallback =
+      jsg::Function<void(jsg::Optional<kj::StringPtr>, jsg::Optional<kj::Array<kj::byte>>)>;
+
+  uint32_t crc32Sync(InputSource data, uint32_t value);
+  void zlibWithCb(
+      jsg::Lock& js, InputSource data, Options options, ZlibModeValue mode, CompressCallback cb);
+  kj::Array<kj::byte> zlibSync(InputSource data, Options options, ZlibModeValue mode);
 
   JSG_RESOURCE_TYPE(ZlibUtil) {
     JSG_METHOD_NAMED(crc32, crc32Sync);
     JSG_NESTED_TYPE(ZlibStream);
+    JSG_METHOD(zlibSync);
+    JSG_METHOD(zlibWithCb);
```

Comment:
Let's follow Node.js function naming here to avoid confusion:

sync functions have Sync suffix (zlibSync)
callback functions doesn't have anything (zlib)

```suggestion
    JSG_METHOD(zlib);
```

Reply:
I was a bit scared when something went wrong in edgeworker when you had a function named `crc32`. Maybe I misunderstood the problem?

Reply:
You can use jsg method named and expose different function to JS just like we do for crc32Sync